rules/prefer-event-key.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const keys = ['keyCode', 'charCode', 'which'];
+// https://github.com/facebook/react/blob/b87aabd/packages/react-dom/src/events/getEventKey.js#L36
+// only meta characters which can't be deciphered from String.fromCharCode
+const translateToKey = {
+	8: 'Backspace',
+	9: 'Tab',
+	12: 'Clear',
+	13: 'Enter',
+	16: 'Shift',
+	17: 'Control',
+	18: 'Alt',
+	19: 'Pause',
+	20: 'CapsLock',
+	27: 'Escape',
+	32: ' ',
+	33: 'PageUp',
+	34: 'PageDown',
+	35: 'End',
+	36: 'Home',
+	37: 'ArrowLeft',
+	38: 'ArrowUp',
+	39: 'ArrowRight',
+	40: 'ArrowDown',
+	45: 'Insert',
+	46: 'Delete',
+	112: 'F1',
+	113: 'F2',
+	114: 'F3',
+	115: 'F4',
+	116: 'F5',
+	117: 'F6',
+	118: 'F7',
+	119: 'F8',
+	120: 'F9',
+	121: 'F10',
+	122: 'F11',
+	123: 'F12',
+	144: 'NumLock',
+	145: 'ScrollLock',
+	224: 'Meta'
+};
+
+const isPropertyNamedAddEventListener = node =>
+	node &&
+	node.callee &&
+	node.callee.property &&
+	node.callee.property.name === 'addEventListener';
+
+const getEventNodeAndReferences = (context, node) => {
+	const eventListener = getMatchingAncestorOfType(node, 'CallExpression', isPropertyNamedAddEventListener);
+	const callback = eventListener && eventListener.arguments && eventListener.arguments[1];
+	switch (callback && callback.type) {
+		case 'ArrowFunctionExpression':
+		case 'FunctionExpression': {
+			const eventVariable = context.getDeclaredVariables(callback)[0];
+			const references = eventVariable && eventVariable.references;
+			return {
+				event: callback.params && callback.params[0],
+				references
+			};
+		}
+
+		default:
+			return {};
+	}
+};
+
+const isPropertyOf = (node, eventNode) => {
+	return (
+		node &&
+		node.parent &&
+		node.parent.object &&
+		node.parent.object === eventNode
```

Comment:
Please also check explicitly for node types (`node.type`, `node.parent.type`, etc.). This also applies to other checks above and below. This is necessary as multiple node types (both currently and in the future) may have a same property with different meaning (example: `object` property is defined on both `MemberExpression` and `WithStatement`, see [estree](https://github.com/estree/estree))

Reply:
@ankeetmaini Is this done?

Reply:
done